src/Console/Command/ObfuscateData.php

```diff
@@ -0,0 +1,225 @@
+        }
+        if ($overwrittenCount == $resultsCount) {
+            $this->info("updated ".$overwrittenCount." records.\n");
+            return $overwrittenCount;
+        } else {
+            $this->info("failed to overwrite all records.\n");
+            return false;
+        }
+    }
+
+    private function overwritePostAuthors()
+    {
+        $this->info("Overwriting post author data...");
+        $faker = Faker\Factory::create();
+        $results = DB::select('posts.*')
+            ->from('posts')
+            ->execute($this->db)
+            ->as_array();
+        $overwrittenCount = 0;
+        $resultsCount = count($results);
+        foreach ($results as $row) {
+            $postEntity = $this->postRepository->getEntity($row);
+            $fakeEmail = $faker->safeEmail;
+            $fakeName = $faker->firstName." ".$faker->lastName;
+            $postEntity->setState([
+                'author_realname'=> $fakeName,
+                'author_email'=> $fakeEmail,
+            ]);
+            $overwrittenCount += $this->postRepository->update($postEntity);
+        }
+        if ($overwrittenCount == $resultsCount) {
+            $this->info("updated ".$overwrittenCount." records.\n");
+            return $overwrittenCount;
+        } else {
+            $this->info("failed to overwrite all records.\n");
+            return false;
+        }
+    }
+
+    private function overwriteSiteConfig()
+    {
+        $this->info("\nORemoving general settings data...");
+        $siteConfig = $this->configRepository->get('site');
+        $siteConfig->setState([
+            'name' => 'Test site',
+            'description' => 'Staging deployment for testing',
+            'email' => 'admin@ushahidi.com',
+            //TODO: reset these?
+            //'cient_url' => '??',
+            // 'url' => '??'
+        ]);
+        $this->configRepository->update($siteConfig);
+        $this->info("\nDone.");
+    }
+
+    private function removeDataProviderValues()
+    {
+        $this->info("\nORemoving dataProvider data...");
+        $dataProviderConfig = $this->configRepository->get('data-provider');
+        //TODO: walk through each key instead?
+        //TODO:  otherwise set them to defaults?
+        $dataProviderConfig->setState([
+            'url' => '',
+            'providers' => [
+                'smssync' => false,
+                'email' => false,
+                'twilio' => false,
+                'nexmo' => false,
+                'twitter' => false,
+                'frontlinesms' => false
+            ],
+            'frontlinesms'=> [],
+            'nexmo'=> [],
+            'smssync'=> [],
+            'twilio'=> [],
+            'email'=> [],
+        ]);
+        $this->configRepository->update($dataProviderConfig);
+        $this->info("\nSet all dataproviders to false.");
+    }
+
+    private function deleteUsers()
+    {
+        $this->info("Deleting users...");
+        $query = DB::delete('users');
```

Comment:
I think we should replace users with junk data, rather than wiping them outright. It makes for much more useful test data ie. many users not just a single user.

Reply:
True. I'm trying to decide if we should then just scramble passwords, or set them all to one obvious thing.

Reply:
Left this method in place, in case we want to add this as an option.

Reply:
I think scramble passwords unless its too slow to work. In which case set them all to the same things but make it long and random